### PR TITLE
Add build callback tests

### PR DESCRIPTION
**Pull request by irenedea**
Adds tests to check that we can build callbacks correctly.

### CODE REVIEW

File: tests/callbacks/test_callbacks.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+from composer.core import Callback
+
+from llmfoundry.registry import callbacks, callbacks_with_config
+from llmfoundry.utils.builders import build_callback
+
+
+def test_build_callbacks():
+    entry_points = callbacks.get_entry_points()
```
Comment (dakinggg) [inline tests/callbacks/test_callbacks.py:11]: I think you want to use this https://github.com/mosaicml/llm-foundry/blob/23caea7532fb37f23fde2a1bfff798f565f9aeb2/llmfoundry/utils/registry_utils.py#L69?

Comment (irenedea) [reply tests/callbacks/test_callbacks.py:11]: oops, yeah you're right. What are entry points?

Comment (dakinggg) [reply tests/callbacks/test_callbacks.py:11]: entry points are just the part of the registry specified via pyproject.toml

File: tests/callbacks/test_callbacks.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+
+def get_default_kwargs(callback_class: type):
+    type_hints = typing.get_type_hints(callback_class.__init__)
+    inspected_params = inspect.signature(callback_class.__init__).parameters
+
+    default_kwargs = {}
+
+    for param, tpe in type_hints.items():
+        print(param, tpe)
```
Comment (dakinggg) [inline tests/callbacks/test_callbacks.py:65]: rm

**Review by dakinggg — APPROVED** [review]

File: tests/callbacks/test_callbacks.py (hunk 2 above)
Comment (irenedea) [reply tests/callbacks/test_callbacks.py:65]:
```suggestion
```

**Pull request merged**